static/components/List.js

```diff
@@ -1,78 +1,85 @@
 import React, { Component } from 'react';
-import ListItem from './listitem';
-import uuidv4 from 'uuid/v4';
+import ListItem from './ListItem';
+import uuidv4 from 'uuid/v4'; // generate unique id
 
 class List extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            items: []
+            items: [],
         };
-        this.removeItem = this.removeItem.bind(this)
+        this.removeItem = this.removeItem.bind(this);
+        this.editItem = this.editItem.bind(this);
     }
 
-    onSubmit = (event) => {
-        event.preventDefault();
-        const newValue = this.input.value;
-        
-        if(newValue == '') {
-            return false;
-        } else {
-            var newItem = {
-                text: newValue,
-                date: new Date().toLocaleTimeString(),
-                id: uuidv4()
-            }
-            var newItems = [...this.state.items, newItem];
+    addTodo = (props) => {
+        let newItem = {};
+        let newItems = [];
 
+        if(props.id == '') {
+            newItem = {
+                text: props.term,
+                date: new Date().getSeconds(),
+                id: uuidv4(),
+            }
+            newItems = [...this.state.items, newItem];
             this.setState({
-                items: newItems
+                items: newItems,
             });
-            
-            this.input.value = '';
+        }
+        else {
+            this.state.items.forEach(elem => {
+                if (elem.id == props.id) {
+                    elem.text = props.term;
+                }
+            })
         }
     }
 
-    removeItem (item) {
-        let filterItems = this.state.items.filter((eq) => {
-            return eq.id !== item.id
+    editItem = (item) => {
+        this.props.onEdit(item);
+    }
+
+    removeItem = (item) => {
+        let filterItems = this.state.items.filter((elem) => {
+            return elem.id !== item.id;
+        });
+
+        this.setState({
+            items: filterItems,
         });
+    }
 
+    sort = () => {
+        function sortItems (itemA, itemB) {
+            return itemB.date - itemA.date
+        }
+
+        let sortArray = this.state.items.sort(sortItems);
```

Comment:
This is a bad implementation, as Array.prototype.sort actually mutates array, which is not allowed with state